python/vast/vast.py

```diff
@@ -0,0 +1,164 @@
+import asyncio
+import io
+from collections import defaultdict
+
+from dynaconf import Dynaconf
+import pyarrow as pa
+import stix2
+
+from .fabric import Fabric
+import vast.backbones
+import vast.bridges.stix
+import vast.utils.arrow
+import vast.utils.logging
+
+logger = vast.utils.logging.get("node")
+
+
+class CLI:
+    """A commmand-line wrapper for the VAST executable."""
+
+    @staticmethod
+    def arguments(**kwargs):
+        result = []
+        for k, v in kwargs.items():
+            if v is True:
+                result.append(f"--{k.replace('_','-')}")
+            else:
+                result.append(f"--{k.replace('_','-')}={v}")
+        return result
+
+    def __init__(self, **kwargs):
+        self.args = CLI.arguments(**kwargs)
+
+    async def exec(self, stdin=False):
+        async def run(*args, stdin):
+            return await asyncio.create_subprocess_exec(
+                "vast",
+                *args,
+                stdin=asyncio.subprocess.PIPE if stdin else None,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+        logger.debug(f"> vast {' '.join(self.args)}")
+        if not stdin:
+            return await run(*self.args, stdin=False)
+        proc = await run(*self.args, stdin=True)
+        proc.stdin.write(str(stdin).encode())
+        await proc.stdin.drain()
+        proc.stdin.close()
```

Comment:
I guess you took this out of pyvast, but I think this might lead to trouble if your command streams large inputs into large outputs as you are waiting for all of stdin to be buffered in before starting to read the output

Reply:
I've added a comment to look out for this. Does that work for now?